template/golang-middleware/Dockerfile

```diff
@@ -17,16 +17,22 @@ RUN mkdir -p /go/src/handler
 WORKDIR /go/src/handler
 COPY . .
 
+ARG GO111MODULE="off"
+ARG GOPROXY=""
+ARG GOFLAGS=""
+
 # Add user overrides to the root go.mod, which is the only place "replace" can be used
 RUN cat function/GO_REPLACE.txt >> ./go.mod || exit 0
+RUN if [ -d ./function/vendor ] && [ "${GO111MODULE}" == "off" ]; then \
```

Comment:
As a note, I don't think we'll need the `GO111MODULE` check in 1.17, when it can't be turned off.